src/main/java/org/apache/commons/io/output/AbstractByteArrayOutputStream.java

```diff
@@ -92,7 +92,7 @@ protected interface InputStreamConstructor<T extends InputStream> {
     private byte[] currentBuffer;
 
     /** The index of the current buffer. */
-    private int currentBufferIndex;
+    private int currentBufferIndex = -1;
```

Comment:
unrelated to the bug but I found this value was wrong while investigating. I dont think its causing any issues because it appears to only be used after `reset` is called which properly sets this value.